package/base-files/files/bin/board_detect

```diff
@@ -1,14 +1,20 @@
 #!/bin/sh
 
-CFG=$1
+CFG="${1:-/etc/board.json}"
 
-[ -n "$CFG" ] || CFG=/etc/board.json
-
-[ -d "/etc/board.d/" -a ! -s "$CFG" ] && {
-	for a in $(ls /etc/board.d/*); do
-		[ -s $a ] || continue;
-		$(. $a)
+if [ -d '/etc/board.d/' ] && \
+   [ ! -s "${CFG}" ]; then
+	for _script in '/etc/board.d/'*; do
+		if [ ! -s "${_script}" ]; then
```

Comment:
this continue sound like it can be removed.

Reply:
If glob matches nothing it will return the glob pattern verbatim, I suppose the `-s` check guards against that.

Reply:
right my thinking is:

```
[ -s "${_script}"  && $(. "${_script}")
```

Reply:
> right my thinking is:
> 
> ```
> [ -s "${_script}"  && $(. "${_script}")
> ```

Your reasoning is sound, but it also causes unnecessary nesting, and kinda rubs against the pattern 'fail early, fail often' E.g. check for your error condition, and fail/abort (continue in this case),